---
title: "Differential Gene Expression of B. Thetaiotamicron"
author: "Gina Vazquez"
date: "2023-11-09"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r, echo=FALSE, include=FALSE}
library(SummarizedExperiment)
library(tximeta)
library(DESeq2)
library(dplyr)
library(ggplot2)
library(readr)
library(pheatmap)
library(tidyverse)
library(IHW)
library(ashr)
library(ggplot2)
```

## Background
RNA-Sequencing data for Zhu et al., 2020 and Spiga et al., 2017 were collected at the same time. Here I am comparing the gene expression of Bacteriodes Thetaiotamicron in two groups. Samples 275.1 - 275.5 are of a mouse gut (GRCm39) inoculated with Clostridium Symbiosum and Bacteriodes Thetaiotamicron VPI-5482. Samples 278.1 - 278-5 are of a mouse gut (GRCm39) inoculated with Clostridium Symbiosum, Bacteriodes Thetaiotamicron VPI-5482, and Salmonella Typhimurium. 

### Prepping FeatureCounts table for Deseq2

```{r}
df_bth_fc = read.delim("bth_counts_1109.txt", skip = 1, header = TRUE, row.names = 1, stringsAsFactors = FALSE)
df_bth = df_bth_fc[c("X275.1_bth.sorted.bam",	"X275.2_bth.sorted.bam",	"X275.3_bth.sorted.bam",	"X275.4_bth.sorted.bam",	"X275.5_bth.sorted.bam",	"X278.1_bth.sorted.bam",	"X278.2_bth.sorted.bam",	"X278.3_bth.sorted.bam",	"X278.4_bth.sorted.bam",	"X278.5_bth.sorted.bam")]

head(df_bth_fc)
head(df_bth)
```

```{r}
exp_bth = data.frame(
  Column1 = c("X275.1_bth.sorted.bam",	"X275.2_bth.sorted.bam",	"X275.3_bth.sorted.bam",	"X275.4_bth.sorted.bam",	"X275.5_bth.sorted.bam",	"X278.1_bth.sorted.bam",	"X278.2_bth.sorted.bam",	"X278.3_bth.sorted.bam",	"X278.4_bth.sorted.bam",	"X278.5_bth.sorted.bam"),
  Column2 = c("cs/bt", "cs/bt", "cs/bt", "cs/bt", "cs/bt", "cs/bt/stm", "cs/bt/stm", "cs/bt/stm", "cs/bt/stm", "cs/bt/stm" )
)
colnames(exp_bth) = c("sample", "condition")
exp_bth_path = "~/Documents/Prelim_GMMs/exp_bth.csv"
write.csv(exp_bth, file = exp_bth_path, row.names = FALSE)
csv_exp_bth = read.csv("exp_bth.csv")

csv_exp_bth
```

### DeSeq2 Object
I used the CSV with the experimental conditions and featurecounts table to make the DeSeq2 Object. 

```{r, include=FALSE}
dds_bth = DESeqDataSetFromMatrix(countData = df_bth, colData = csv_exp_bth, design = ~ condition)
```

### Results
Before looking at the results. I filtered out samples with counts less than 10 and set the factor level so that the cs/bt/stm group is being compared to the "mock" group cs/bt. Note on factor levels - alternatively, they can be set by using the contrast argument with *results*.

```{r}
#keeping only genes with more than 10 counts
keep_bth = rowSums(counts(dds_bth)) >= 10
dds_bth = dds_bth[keep_bth, ]

#setting factor levels
dds_bth$condition = factor(dds_bth$condition, levels = c("cs/bt","cs/bt/stm"))

#deseq2 results 
dds_bth = DESeq(dds_bth)
bth_res = results(dds_bth)

head(bth_res)
```

MA Plot of BTH group DGE w/o shrinkage

```{r, echo=FALSE}
ma_data_bth = data.frame(M = bth_res$log2FoldChange, A = rowMeans(counts(dds_bth, normalized = TRUE)))
ggplot(ma_data_bth, aes(y = M, x = A)) +
  geom_point() +
  labs(y = "Log2 Fold Change (M)", x = "Average Expression (A)") +
  scale_x_continuous(trans='log10') +
  ggtitle("MA Plot for BTH Set")
```
#### Different types of shrinkage and how it affects the MA Plot for DGE

```{r, include=FALSE}
apeglm_bth = lfcShrink(dds_bth, coef="condition_cs.bt.stm_vs_cs.bt", type="apeglm")

norm_bth = lfcShrink(dds_bth, coef=2, type="normal")

ashr_bth = lfcShrink(dds_bth, coef=2, type="ashr")
```

```{r, echo=FALSE}
plotMA(apeglm_bth, main="apeglm")
plotMA(norm_bth, main="norm")
plotMA(ashr_bth, main="ashr")
```

```{r}
vsd_bth = vst(dds_bth)
```