---
title: "Differential Gene Expression of S. Tm"
author: "Gina Vazquez"
date: "2023-11-09"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r, echo=FALSE, include=FALSE}
library(SummarizedExperiment)
library(tximeta)
library(DESeq2)
library(dplyr)
library(ggplot2)
library(readr)
library(pheatmap)
library(tidyverse)
library(IHW)
library(ashr)
library(ggplot2)
```

## Background
RNA-Sequencing data for Zhu et al., 2020 and Spiga et al., 2017 were collected at the same time. Here I am comparing the gene expression of Salmonella Typhimurium in two groups. Samples 278.1 - 278-5 are of a mouse gut (GRCm39) inoculated with Clostridium Symbiosum, Bacteriodes Thetaiotamicron VPI-5482, and Salmonella Typhimurium. Samples 279.1 - 279.5 are of a mouse gut (GRCm39) inoculated with Salmonella Typhimurium. 

### Prepping FeatureCounts table for Deseq2

```{r}
df_stm_fc = read.delim("stm_counts_1109.txt", skip = 1, header = TRUE, row.names = 1, stringsAsFactors = FALSE)
df_stm = df_stm_fc[c("X278.1_stm.sorted.bam",	"X278.2_stm.sorted.bam",	"X278.3_stm.sorted.bam",	"X278.4_stm.sorted.bam",	"X278.5_stm.sorted.bam",	"X279.1_stm.sorted.bam",	"X279.2_stm.sorted.bam",	"X279.3_stm.sorted.bam",	"X279.4_stm.sorted.bam",	"X279.5_stm.sorted.bam")]

head(df_stm_fc)
head(df_stm)
```

```{r}
exp_stm = data.frame(
  Column1 = c("X278.1_stm.sorted.bam",	"X278.2_stm.sorted.bam",	"X278.3_stm.sorted.bam",	"X278.4_stm.sorted.bam",	"X278.5_stm.sorted.bam",	"X279.1_stm.sorted.bam",	"X279.2_stm.sorted.bam",	"X279.3_stm.sorted.bam",	"X279.4_stm.sorted.bam",	"X279.5_stm.sorted.bam"), 
  Column2 = c("cs/bt/stm", "cs/bt/stm", "cs/bt/stm", "cs/bt/stm", "cs/bt/stm", "stm", "stm", "stm", "stm", "stm")
)
colnames(exp_stm) = c("sample", "condition")
exp_stm_path = "~/Documents/Prelim_GMMs/exp_stm.csv"
write.csv(exp_stm, file = exp_stm_path, row.names = FALSE)
csv_exp_stm = read.csv("exp_stm.csv")

csv_exp_stm
```

### DeSeq2 Object
I used the CSV with the experimental conditions and featurecounts table to make the DeSeq2 Object.

```{r, include=FALSE}
dds_stm = DESeqDataSetFromMatrix(countData = df_stm, colData = csv_exp_stm, design = ~ condition)
```

### Results
Before looking at the results. I filtered out samples with counts less than 10 and set the factor level so that the cs/bt/stm group is being compared to the "mock" group stm. Note on factor levels - alternatively, they can be set by using the contrast argument with *results*.

```{r}
#keeping only genes with more than 10 counts
keep_stm = rowSums(counts(dds_stm)) >= 10 
dds_stm = dds_stm[keep_stm, ]

#setting factor levels
dds_stm$condition = factor(dds_stm$condition, levels = c("stm","cs/bt/stm"))

#deseq2 results 
dds_stm = DESeq(dds_stm)
stm_res = results(dds_stm)

head(stm_res)
```

MA Plot of STM group DGE w/o shrinkage

```{r, echo=FALSE}
ma_data_stm = data.frame(M = stm_res$log2FoldChange, A = rowMeans(counts(dds_stm, normalized = TRUE)))
ggplot(ma_data_stm, aes(x = A, y = M)) +
  geom_point() +
  labs(y = "Log2 Fold Change (M)", x = "Average Expression (A)") + 
  xlim(c(.1, 1e4)) +
  scale_x_continuous(trans='log10') +
  ggtitle("MA Plot for STM Set")
```

#### Different types of shrinkage and how it affects the MA Plot for DGE

```{r, include=FALSE}
apeglm_stm = lfcShrink(dds_stm, coef="condition_cs.bt.stm_vs_stm", type="apeglm")

norm_stm = lfcShrink(dds_stm, coef=2, type="normal")

ashr_stm = lfcShrink(dds_stm, coef=2, type="ashr")
```

```{r, echo==FALSE}
plotMA(apeglm_stm, main="apeglm")
plotMA(norm_stm, main="norm")
plotMA(ashr_stm, main="ashr")
```

```{r}
vsd_stm = vst(dds_stm)
```

```{r}
#sample distances
sample_dists_stm = assay(vsd_stm) %>%
  t() %>%
  dist() %>%
  as.matrix()
head(sample_dists_stm)

#MDS values from distance matrix
mds_data_stm = data.frame(cmdscale(sample_dists_stm))
mds_stm = cbind(mds_data_stm, as.data.frame(colData(vsd_stm)))
head(mds_stm)
```

```{r, echo=FALSE}
ggplot(mds_stm, aes(X1, X2, shape = condition)) + 
  geom_point(size = 3) +
  theme_minimal()
```
